src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1873,4 +1881,101 @@ private boolean expectedFunction(@Nullable PsiElement resolve) {
+    }
+
+    @Nullable
+    private static String getParameterName(@NotNull ErlangFunctionClause clause, int position) {
+      List<ErlangArgumentDefinition> definitions = clause.getArgumentDefinitionList().getArgumentDefinitionList();
+      ErlangArgumentDefinition definition = definitions.size() > position ? definitions.get(position) : null;
+      ErlangMaxExpression expression = ObjectUtils.tryCast(definition != null ? definition.getExpression() : null,
+                                                           ErlangMaxExpression.class);
+      ErlangQVar var = expression != null ? expression.getQVar() : null;
+      return var != null ? var.getName() : null;
+    }
+  }
+
+  public static class ErlangFunctionCallModuleParameter extends PatternCondition<ErlangQAtom> {
+    public ErlangFunctionCallModuleParameter() {
+      super("functionCallModuleParameter");
+    }
+
+    @Override
+    public boolean accepts(@NotNull ErlangQAtom atom, ProcessingContext context) {
+      ErlangMaxExpression expr = ObjectUtils.tryCast(atom.getParent(), ErlangMaxExpression.class);
+      ErlangArgumentList args = ObjectUtils.tryCast(expr != null ? expr.getParent() : null, ErlangArgumentList.class);
+      ErlangSpecification specification = getSpecification(args);
+      if (expr == null || args == null || specification == null) return false;
+
+      int position = args.getExpressionList().indexOf(expr);
+      if (position == -1) return false;
+
+      ErlangFunTypeSigs signature = specification.getSignature();
+      if (signature == null) return false;
+
+      for (ErlangTypeSig typeSig : signature.getTypeSigList()) {
+        List<ErlangType> types = typeSig.getFunType().getFunTypeArguments().getTypeList();
+        if (types.size() <= position) continue;
+        ErlangType type = types.get(position);
+        ErlangQVar typeVar = type.getQVar();
+        if (argumentIsModule(typeVar != null ? getType(typeVar, typeSig.getTypeSigGuard()) : type)) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    @Nullable
+    private static ErlangType getType(@NotNull ErlangQVar typeVar, @Nullable ErlangTypeSigGuard sigGuard) {
+      if (sigGuard == null) return null;
+      for (ErlangTypeGuard guard : sigGuard.getTypeGuardList()) {
+        for (ErlangType topType : guard.getTypeList()) {
+          ErlangQVar var = topType.getQVar();
+          if (var != null && typeVar.getName().equals(var.getName()) && topType.getType() != null) {
+            return topType.getType();
+          }
+        }
+      }
+      return null;
+    }
+
+    private static boolean argumentIsModule(@Nullable ErlangType type) {
+      ErlangTypeRef ref = type != null ? type.getTypeRef() : null;
+      ErlangAtom atom = ref != null ? ref.getQAtom().getAtom() : null;
+      return atom != null && "module".equals(atom.getName());
+    }
+
+    @Nullable
+    private static ErlangSpecification getSpecification(@Nullable ErlangArgumentList args) {
```

Comment:
Ideologically, this method is bad. It is to find ErlangSpecification given ErlangArgumentList. ErlangArgumentList is used not only in function calls, but also in other places. It is misleading to have a method like this.